.github/workflows/main.yml

```diff
@@ -37,4 +37,4 @@ jobs:
           node-version: 18
       - run: npm install
       - run: npm install typescript@${{ matrix.typescript-version }}
-      - run: npx tsc
+      - run: npm test
```

Comment:
This makes it test more than just typescript and may fail for unrelated reasons

Reply:
As was mentioned in the issue, `tsd` ships with its own copy of typescript and as such isn't affected by any changes to the `typescript` dependency: https://github.com/sindresorhus/type-fest/issues/831#issuecomment-1995912221

One would instead have to do what I believe @shicks and @trevorade are doing: Use eg. [`expect-type`](https://www.npmjs.com/package/expect-type) instead: https://github.com/sindresorhus/type-fest/pull/499

And since `expect-type` is validated as part of running `tsc` itself, no changes to the GitHub Actions would be done then either.

That said: Replacing `tsd` with `expect-type` is a much broader topic, as highlighted by eg https://github.com/sindresorhus/type-fest/pull/499

Though: Since `expect-type` is already a dev dependency  of `type-fest` as far as I remember it could be good to add a test using that and ensure that breaks the canary tests correctly, at least that way we won't see a regression again.

Reply:
Got it

Reply:
> Though I'm a bit uncertain if those are actually run using `tsc` rather than `tsd`

Unfortunately it's using `tsd`.

Directory `test-d/` is excluded in tsconfig. https://github.com/sindresorhus/type-fest/blob/4f14bff7321b9a9876dca0719945423abd6b4da1/tsconfig.json#L14C2-L14C16